### PR TITLE
[keycloak] Adding externalTrafficPolicy to service definitions

### PR DESCRIPTION
**Pull request by robcoward**
This PR allows the deployment to explicitly set the `externalTrafficPolicy` when deploying a service type 'LoadBalancer', allowing the default value of `Cluster` to be changed to `Local` as part of the chart provisioning of the load balanced service.

### CODE REVIEW

File: charts/keycloak/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: keycloak
-version: 9.9.5
+version: 9.9.6
```
Comment (unguiculus) [inline charts/keycloak/Chart.yaml:3]: Please bump the minor version for a new feature to follow semver.
```suggestion
version: 9.10.0
```

File: charts/keycloak/values.schema.json [hunk 2]
```diff
@@ -431,4 +431,4 @@
       }
     }
   }
-}
+}
```
Comment (unguiculus) [inline charts/keycloak/values.schema.json:434]: Keep the line break
```suggestion
}

```

File: charts/keycloak/templates/service-http.yaml [hunk 3]
```diff
@@ -23,6 +23,9 @@ spec:
   loadBalancerSourceRanges:
     {{- toYaml .Values.service.loadBalancerSourceRanges | nindent 4 }}
   {{- end }}
+  {{- if and (eq "LoadBalancer" .Values.service.type) .Values.service.externalTrafficPolicy }}
+  externalTrafficPolicy: {{ .Values.service.loadBalancerSourceRanges }}
+  {{- end }}
```
Comment (unguiculus) [inline charts/keycloak/templates/service-http.yaml:28]: Move default value to values.yaml
```suggestion
  {{- if eq "LoadBalancer" .Values.service.type }}
  externalTrafficPolicy: {{ .Values.service.loadBalancerSourceRanges }}
  {{- end }}
```

File: charts/keycloak/values.yaml [hunk 4]
```diff
@@ -242,6 +242,10 @@ service:
   # to connect to the LoadBalancer, e. g. will result in Security Groups
   # (or equivalent) with inbound source ranges allowed to connect
   loadBalancerSourceRanges: []
+  # When using Service type LoadBalancer, you can preserve the source IP seen in the container
+  # by changing the default (Cluster) to be Local.
+  # See https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip
+  externalTrafficPolicy: ""
```
Comment (unguiculus) [inline charts/keycloak/values.yaml:248]: Add default value here
```suggestion
  externalTrafficPolicy: Cluster
```

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please update the readme as well.

File: charts/keycloak/Chart.yaml (hunk 1 above)
Comment (robcoward) [reply charts/keycloak/Chart.yaml]: minor version bumped as requested

File: charts/keycloak/templates/service-http.yaml (hunk 3 above)
Comment (robcoward) [reply charts/keycloak/templates/service-http.yaml]: Updated as requested

File: charts/keycloak/values.yaml (hunk 4 above)
Comment (robcoward) [reply charts/keycloak/values.yaml]: Updated as requested

**Pull request closed without merging**